### PR TITLE
Fix #5147: 🐛 Fix navigation issue with disabled dates in date picker

### LINKED ISSUE #5147: Keyboard Navigation to the days of a month is not working if the pre-selected date is disabled
**Describe the bug**

If we select a date and disable the same date in the next or previous months, then when we navigate to the corresponding month and try using arrow keys (Tab) to navigate to the date, it won't work as there is no pre-selected date available in the month due to the fact that the corresponding date we selected previously is disabled in the corresponding month. 

**To Reproduce**
```
() => {
  const [startDate, setStartDate] = useState(new Date("2024-06-01"));
  return (
    <DatePicker
      selected={startDate}
      onChange={(date) => setStartDate(date)}
      excludeDates={[new Date("2024-07-01")]}
    />
  );
};
```
In the above example, I programmatically selected 1st June, 2024 and disabled 1st July, 2024 (same day in the next month).  Here once we navigate to July Month using the next month arrow keys, we can't able to switch the focus back any days of the month.

**Expected behavior**
It should pre-select the first enabled date in a month in case of the previously selected date is disabled in the corresponding month.

### PR DESCRIPTION
**Pull request by balajis-qb**
Closes #5147

**Problem**
As I described in the linked ticket, the issue is if we select a date and disable the same date in the next or previous months, then when we navigate to the corresponding month and try using arrow keys (Tab) to navigate to the date, it won't work as there is no pre-selected date available in the month due to the fact that the corresponding date we selected previously is disabled in the corresponding month.

**Changes**
- Whenever month gets switched, instead of auto-preselecting the already selected day from the selected date.  I first checked if the same day in the month is disabled, if not pre-select the same day, else pre-select the first enabled date in the month.  If all the days are disabled in the rendered month, didn't pre-select any days.

## Contribution checklist
- [x] I have followed the [contributing guidelines](https://github.com/Hacker0x01/react-datepicker/blob/main/CONTRIBUTING.md).
- [x] I have added sufficient test coverage for my changes.
- [x] I have formatted my code with Prettier and checked for linting issues with ESLint for code readability.

### CODE REVIEW

File: src/test/calendar_test.test.tsx [hunk 1]
```diff
@@ -2257,6 +2260,108 @@ describe("Calendar", () => {
...
+        const selected = new Date("2024-06-01");
+        const excludeDate = addMonths(selected, 1);
+
+        const { container } = render(
+          <DatePicker selected={selected} excludeDates={[excludeDate]} />,
+        );
+        const input = safeQuerySelector<HTMLInputElement>(container, "input");
+        fireEvent.focus(input);
+
+        const nextButton = container.querySelector(
```
[the remark this answers is not in the document]
Comment (balajis-qb) [reply src/test/calendar_test.test.tsx]: Thank you for the catch.  Sorry, I accidentally used it in one test case and copied the same to the other test cases.  I fixed it now.

**Comment by martijnrusschen** [comment]
The test suite is failing, please check on any issues.

**Comment by balajis-qb** [comment]
Hi @martijnrusschen , when I check the issue it's saying code formatting issue on calendar.tsx and time.tsx.  But when I tried `yarn prettier:check` in my local it's working.
image

**Comment by balajis-qb** [comment]
@martijnrusschen, I reinstalled the packages and fixed this code formatting issue.  This PR is now safe to be merged.

**Pull request merged**